src/opera/pge/disp_s1/disp_s1_pge.py

```diff
@@ -139,6 +142,188 @@ def _validate_output(self):
+
+        Parameters
+        ----------
+        inter_filename : str, optional
+            The intermediate filename of the output product to generate the
+            core filename for. This parameter may be used to inspect the file
+            in order to derive any necessary components of the returned filename.
+            Once the core filename is cached upon first call to this function,
+            this parameter may be omitted.
+
+        Returns
+        -------
+        core_filename : str
+            The core file name component to assign to products created by this PGE.
+
+        """
+        # Check if the core filename has already been generated and cached,
+        # and return it if so
+        if self._cached_core_filename is not None:
+            return self._cached_core_filename
+
+        # Assign the core file name to the cached class attribute
+        self._cached_core_filename = (
+            f"{self.PROJECT}_{self.LEVEL}_{self.NAME}"
+        )
+
+        return self._cached_core_filename
+
+    def _frame_filename(self, inter_filename=None):
+        """
+        Returns the file name to use for frame-based DISP products produced
+        by this PGE.
+
+        The filename for the DISP-S1 frame products consists of:
+
+        <CoreFilename>_<Mode>_<FrameID>_<Polarization>_<ReferenceDateTime>_<SecondaryDateTime>_<ProductVersion>_<ProductGenerationDateTime>
+
+        Where <CoreFilename> is returned by DispS1PostProcessorMixin._core_filename()
+
+        Callers of this function are responsible for assignment of any other
+        product-specific fields, such as the file extension.
+
+        Parameters
+        ----------
+        inter_filename : str, optional
+            The intermediate filename of the output product to generate the
+            core filename for. This parameter may be used to inspect the file
+            in order to derive any necessary components of the returned filename.
+            Once the core filename is cached upon first call to this function,
+            this parameter may be omitted.
+
+        Returns
+        -------
+        frame_filename : str
+            The filename component to assign to frame-based products created by
+            this PGE.
+        """
+        core_filename = self._core_filename(inter_filename)
+
+        disp_metadata = self._collect_disp_s1_product_metadata(inter_filename)
+
+        # Mode: S1-A/B acquisition mode, should be fixed to "IW"
+        mode = "IW"
+
+        # FrameID: Unique frame identification number as a 5-digit string in
+        # the format FXXXXX
+        frame_id = f"F{disp_metadata['identification']['frame_id']:05d}"
+
+        # Polarization:  polarization of the input bursts
+        # derived from names of input CSLCs in the runconfig
+        cslc_file_list = self.runconfig.sas_config['input_file_group']['cslc_file_list']
+        pattern = re.compile(r"t\w{3}_\d{6}_iw[1|2|3]_[0-9]{8}_(VV|VH).h5")
```

Comment:
To ensure we've covered all bases, the list of possible polarization values should be `VV|VH|HH|HV|VV\+VH|HH\+HV`